miden-tx/src/executor/mod.rs

```diff
@@ -38,10 +38,7 @@ pub struct TransactionExecutor {
     authenticator: Option<Arc<dyn TransactionAuthenticator>>,
     /// Holds the code commitments of all accounts loaded into this transaction executor via the
     /// [Self::load_account_code()] method.
-    ///
-    /// These commitments are used to create the account procedure index map during transaction
-    /// execution.
-    account_code_commitments: BTreeSet<Digest>,
+    account_codes: BTreeMap<Digest, AccountCode>,
```

Comment:
I think this could probably be `BTreeSet<AccountCode>` as I don't think we actually use this as a map in the code below.

Reply:
I couldn't use `AccountCode` because it's not `Ord`, but I can implement it based on the commitment if that sounds good.